lib/fuzzy-finder-view.js

```diff
@@ -390,27 +390,32 @@ module.exports = class FuzzyFinderView {
     return {uri: filePath, filePath, label}
   }
 
-  getAbsolutePath (relativePath) {
-    const directories = atom.project.getDirectories()
+  parseResultsFromFuzzyNative (results) {
+    const absolutePaths = new Set()
 
-    if (directories.length === 1) {
-      return path.join(directories[0].getPath(), relativePath)
-    }
+    for (const {value: relativePath} of results) {
+      const directories = atom.project.getDirectories()
 
-    // Remove the first part of the relative path, since it contains the project
-    // directory name if there are many directories opened.
-    relativePath = path.join(...relativePath.split(path.sep).slice(1))
+      if (directories.length === 1) {
+        absolutePaths.add(path.join(directories[0].getPath(), relativePath))
+      } else {
+        // Remove the first part of the relative path, since it contains the project
+        // directory name if there are many directories opened.
+        const newRelativePath = path.join(...relativePath.split(path.sep).slice(1))
 
-    for (const directory of directories) {
-      const absolutePath = path.join(directory.getPath(), relativePath)
+        for (const directory of directories) {
+          const absolutePath = path.join(directory.getPath(), newRelativePath)
 
-      if (fs.existsSync(absolutePath)) {
-        return absolutePath
+          if (fs.existsSync(absolutePath)) {
```

Comment:
I realize that we this `existsSync` call wasn't introduced with this change, but I really think we should avoid performing synchronous I/O on the main thread. You never know how long the disk read is going to block for, especially on a network drive. One way to do this would be to retain the absolute path of the results returned from `fuzzy-native`. I'm not familiar with its current API, so this may require changes to that module. I doubt they would be too difficult and I am happy to help out with it.

Reply:
Good point! I briefly looked at using the async version but that would require changes to `atom-select-list` since it currently expects the `filterFn` function [to be sync](https://github.com/atom/atom-select-list/blob/master/src/select-list-view.js#L293), and changing that was not trivial.

I think that what you mention is similar to the alternate design on the description: that can definitely be done and in general terms it would be a better solution, we just need to make sure that the additional payload that we're serializing in order to pass it to the native module does not regress the overall filtering performance.

Reply:
Oh, I *totally* missed your comments in the alternate design section. Sorry about that. I think it's worth pursuing that route fairly soon.